### PR TITLE
Updated similarity based saliency notebook to track Classifier based Saliency

### PR DESCRIPTION
**Pull request by vbhavank**
The following PR changes the similarity-based saliency map demo notebook to better track the classifier-based notebook with positive and negative saliency support.

### CODE REVIEW

**Review by brianhhu — APPROVED** [review]
Some additional comments:

1) We may need to update this notebook to account for the PIL->numpy interface change if that gets merged into master.

2) The RISE-based perturbation for SimilarityScoring doesn't seem to produce meaningful results- it seems to show only negative saliency or only positive self-similarity saliency. We should potentially consider removing it.

3) @Purg and I have discussed that our notebooks are becoming a bit too "similar" to the base OcclusionSaliency one in that they all revolve around the concept of an app which handles the plotting/visualization (this also includes my proposed VIAME/MONAI ones). Each notebook should clearly highlight what capability is being demoed.

File: docs/release_notes/pending_release.rst [hunk 1]
```diff
@@ -20,6 +20,9 @@ Implementations
 * Add ``RISEScoring`` implementation, with the ability to also compute a
   de-biased form of RISE with an optional input parameter.
 
+* Updated ``SimilarityScoring`` to avoid saliency map clipping for supporting
```
[the remark this answers is not in the document]
Comment (Purg) [reply docs/release_notes/pending_release.rst:23]: If the original implementation of SBSM is intentionally constrained to the `[0,1]` range due to only trying to explain regions of positive influence, then I am also on the side of the leaving the SBSM impl here alone.

**Review by Purg — COMMENTED** [review]
Additional comments on the notebook since I apparently cannot make them inline:
- We can probably remove the "Miscellaneous" cell note about `from_config_dict`: it doesn't look like it is used here.
- Technically don't *have* to call `plt.show()`
- This branch will need a rebase and the notebook will need updating for various `PIL`->`numpy` type changes.
- The `pert_masks = np.asarray(pert_masks)` line is not needed any more.
- The "Feature-Extraction Model" description mentions both ResNet18 and ResNet50. Is that intentional? Why are we referencing two different networks?
- I might recommend implementing the `image_loader` and `feat_vec_with_model` parts with torch model instantiation. The combination of these two parts make up the "blackbox" portion of this example.
- Add explanation to the "Calling the Application" section why we are calling the application so many times, e.g. trying to show a->b, b->a and self similarity for both inputs.
  - Maybe just make that an aspect of the "application" for this example?

File: docs/release_notes/pending_release.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (vbhavank) [reply docs/release_notes/pending_release.rst]: Added documentation in the "Calling the application" section around the saliency map not returning negative values.

**Review by Purg — CHANGES_REQUESTED** [review]
Nearly there! LGTM except for an odd addition of `SimilaritySal.npy` to the root of the repo and the probably no longer applicable update to the similarly named file in `tests/data`.

**Review by Purg — APPROVED** [review]

**Pull request merged**